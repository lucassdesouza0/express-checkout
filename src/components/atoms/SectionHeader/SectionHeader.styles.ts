import styled from "@emotion/styled";
import SectionHeader from "./SectionHeader";
import mq from "utils/queries";

export const Header = styled("h2")`
  padding: 20px 0;
  background-color: rgb(255, 231, 131);
  font-size: 16px;
  color: #555;
  text-align: center;
  font-weight: 500;
  box-shadow: 2px 2px 5px 5px #eee;

  ${mq["md"]} {
    padding: 10px 0;
  }
`;
