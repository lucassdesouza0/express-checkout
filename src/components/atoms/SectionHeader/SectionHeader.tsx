import React, { useRef } from "react";

import { Header } from "./SectionHeader.styles";

interface SectionHeaderProps {
  title?: string;
}

const SectionHeader = ({ title }: SectionHeaderProps) => {
  return <Header>{title}</Header>;
};

export default SectionHeader;
