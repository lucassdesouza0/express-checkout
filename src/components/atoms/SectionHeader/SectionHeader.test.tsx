import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import testRender from "../../../tests/render";

import mock from "../../../tests/__mocks__/product";

import SectionHeader from "./SectionHeader";

describe("Button", () => {
  const title = "Header Title";
  test("it should mount", () => {
    testRender(<SectionHeader title={title} />);

    const header = screen.getByText(title);

    expect(header).toBeInTheDocument();
  });

  test("it should match snapshot", () => {
    const { asFragment } = render(<SectionHeader title={title} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
