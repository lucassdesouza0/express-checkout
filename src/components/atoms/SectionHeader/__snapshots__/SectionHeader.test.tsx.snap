// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should match snapshot 1`] = `
<DocumentFragment>
  <h2
    class="css-14jgswk-Header e2we4hg0"
  >
    Header Title
  </h2>
</DocumentFragment>
`;
